src/main/java/simplifier/services/TagServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package simplifier.services;
+
+import com.google.common.collect.Lists;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import simplifier.model.Link;
+import simplifier.model.Tag;
+import simplifier.repositories.TagRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Service
+public class TagServiceImpl implements TagService {
+
+    private TagRepository tagRepository;
+
+    @Autowired
+    public void setTagRepository(TagRepository tagRepository) {
+        this.tagRepository = tagRepository;
+    }
+
+    @Override
+    public List<Tag> saveOrUpdateTags(Link link) {
+        List<Tag> tags = new ArrayList<>();
+
+        for (Tag tag : link.getTags()) {
+            Tag currentTag = tagRepository.findByName(tag.getName());
```

Comment:
Replace findByName() to method which accepts list of tag's names to avoid multiple calls to db in for()
FYI https://stackoverflow.com/questions/18987292/spring-crudrepository-findbyinventoryidslistlong-inventoryidlist-equivalen